core/src/filesystem/media/format/epub.rs

```diff
@@ -89,7 +90,104 @@ impl FileProcessor for EpubProcessor {
 
 	fn process_metadata(path: &str) -> Result<Option<MediaMetadata>, FileError> {
 		let epub_file = Self::open(path)?;
-		Ok(Some(MediaMetadata::from(epub_file.metadata)))
+		let embedded_metadata = MediaMetadata::from(epub_file.metadata);
+
+		// try get opf file
+		let file_path = std::path::Path::new(path).with_extension("opf");
+		if file_path.exists() {
+			// extract OPF data
+			let opf_string = std::fs::read_to_string(file_path)?;
+			let mut reader = Reader::from_str(opf_string.as_str());
+			reader.config_mut().trim_text(true);
+			let mut current_tag = String::new();
+
+			let mut opf_metadata: HashMap<String, Vec<String>> = HashMap::new();
+
+			while let Ok(event) = reader.read_event() {
+				match event {
+					Event::Start(ref e) | Event::Empty(ref e) => {
+						let tag_name =
+							String::from_utf8_lossy(e.name().as_ref()).to_string();
+
+						// normalize tags
+						current_tag = tag_name
+							.strip_prefix("dc:")
+							.unwrap_or(tag_name.as_str())
+							.to_string();
+
+						if let Some(attr) =
+							e.attributes().filter_map(|a| a.ok()).find(|a| {
+								a.key.as_ref() == b"property" || a.key.as_ref() == b"name"
+							}) {
+							current_tag = format!(
+								"{}: {}",
+								current_tag,
+								String::from_utf8_lossy(&attr.value)
+							);
+						}
+					},
+					Event::Text(e) => {
+						let text = e.unescape().unwrap().to_string();
+						opf_metadata
+							.entry(current_tag.clone())
+							.or_default()
+							.push(text);
+					},
```

Comment:
Let's not unwrap, instead maybe something like:

```suggestion
					Event::Text(e) if let Ok(text) = e.unescape() => {
						opf_metadata
							.entry(current_tag.clone())
							.or_default()
							.push(text);
					},
```

Or `let Some(..)` if the result of the function is an option

Reply:
The method you suggested seems to be an experimental rust feature. I have implemented a similar way without the experimental feature.